core/parentchain/block-import-queue/src/block_import_queue.rs

```diff
@@ -0,0 +1,144 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+//! Block import queue implementation
+
+#[cfg(feature = "sgx")]
+use std::sync::SgxRwLock as RwLock;
+
+#[cfg(feature = "std")]
+use std::sync::RwLock;
+
+use crate::{
+	error::{Error, Result},
+	PopFromBlockQueue, PushToBlockQueue,
+};
+use std::{collections::VecDeque, vec::Vec};
+
+/// Block import queue.
+///
+/// Uses RwLock internally to guard against concurrent access and ensure all operations are atomic.
+pub struct BlockImportQueue<PB> {
+	queue: RwLock<VecDeque<PB>>,
+}
```

Comment:
Simple queue for parentchain blocks. Used by the `RetainLatestDispatcher` to queue parentchain blocks for import.